### PR TITLE
DB actions for sending emails set, batch email sending modified

### PR DESCRIPTION
**Pull request by phoongp**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/java/Controller/PDFinfo.java [hunk 1]
```diff
@@ -78,142 +78,142 @@ public String getEmployeeName(String empID) throws Exception{
...
     /**
     * Get average foundation score of an employee
     * @param empID - the employee id of the employee
     * @return a string representation of the average score
     * @throws java.lang.Exception
     */
-	public String getAverageScoresByFoundationEmployeeID(String empID) throws Exception{
-        String SQL = "select avg(e.scores) from Employees_take_Modules e, Modules m where m.category='Foundation' and m.module_id=e.module_id and e.employee_id= :id";
+    public String getAverageScoresByFoundationEmployeeID(String empID) throws Exception{
+        String SQL = "select avg(e.scores) from Employees_take_Modules e, Modules m where m.category_id='FOUND01' and m.module_id=e.module_id and e.employee_id= :id";
```
Comment (kabollinger03) [inline src/java/Controller/PDFinfo.java:125]: Looks like a lot of changes, but apparently we just can't all agree to use spaces instead or tabs, or at least have them autoexpand :P

These are ok changes, but be advised that I *am* refactoring the entire file to reflect that there may be more than the enumerated categories.

File: src/java/Controller/SendEmail.java [hunk 2]
```diff
@@ -57,14 +58,21 @@ protected PasswordAuthentication getPasswordAuthentication() {
...
 
+            
             // Create the message part
             BodyPart messageBodyPart = new MimeBodyPart();
 
             // Now set the actual message
-            messageBodyPart.setText("Attached to this email is the pdf doc of the employee's performance.");
+            messageBodyPart.setText("Congratulations, " + getName(email) + "\nHere is your Performica Report! Inside is the score of each module you completed and a small graph"+
+            " to show where you stand compared to the class average. We hope you enjoyed your experience in our " +
+            " training program and wish you the best for your future!");
```
Comment (kabollinger03) [inline src/java/Controller/SendEmail.java:75]: I feel like there should be some sort of signature and salutation, but this is good enough for now.

File: src/java/Controller/SendEmail.java [hunk 3]
```diff
@@ -74,15 +82,66 @@ protected PasswordAuthentication getPasswordAuthentication() {
...
+            DataSource source = new FileDataSource(path);
             messageBodyPart.setDataHandler(new DataHandler(source));
             messageBodyPart.setFileName(filename);
             multipart.addBodyPart(messageBodyPart);
 
             // Send the complete message parts
-            message.setContent(multipart);
-             Transport tr = session.getTransport("smtp");
+            //message.setContent(multipart);
+            message.setContent(
```
Comment (kabollinger03) [inline src/java/Controller/SendEmail.java:94]: NGL, I have no idea what this does, I'm, just here cause my file was mentioned.

File: src/java/Controller/SendEmail.java [hunk 4]
```diff
@@ -182,60 +243,168 @@ else if(count ==1)
...
     	
     	try
-		{
-			Class.forName("oracle.jdbc.driver.OracleDriver"); // Type 4 Driver Pure Java Driver
-			Connection con = DriverManager.getConnection("jdbc:oracle:thin:@localhost:1521:XE","Student_Performance","Student_Performance");
-			Statement st = con.createStatement();
-			ResultSet rs = st.executeQuery("select employee_id from employees where email = " + "'" + username + "'");
+	{
+            Class.forName("oracle.jdbc.driver.OracleDriver"); // Type 4 Driver Pure Java Driver
+            Connection con = DriverManager.getConnection("jdbc:oracle:thin:@localhost:1521:XE","Student_Performance","Student_Performance");
```
Comment (kabollinger03) [inline src/java/Controller/SendEmail.java:251]: This shit should be in a config file somewhere, with the `class.forname` and setup and stuff run as part of a setup. but absolutely passwords need to be in config so they can be encrypted later... not by up i hope

File: src/java/Controller/SendEmail.java [hunk 5]
```diff
@@ -182,60 +243,168 @@ else if(count ==1)
...
+    	
+    	return " ";
+    }
+    
+    static String getStreamName(String email) {//getEmployee returns an array of attributes
+    	
+    	try
+	{
+            Class.forName("oracle.jdbc.driver.OracleDriver"); // Type 4 Driver Pure Java Driver
+            Connection con = DriverManager.getConnection("jdbc:oracle:thin:@localhost:1521:XE","Student_Performance","Student_Performance");
```
Comment (kabollinger03) [inline src/java/Controller/SendEmail.java:330]: config

File: src/java/Controller/SendEmail.java [hunk 6]
```diff
@@ -182,60 +243,168 @@ else if(count ==1)
...
+	}
     	
     	return " ";
     }
+        static String[] getStartEnd(String email) {//getEmployee returns an array of attributes
+    	
+    	try
+	{
+            Class.forName("oracle.jdbc.driver.OracleDriver"); // Type 4 Driver Pure Java Driver
+            Connection con = DriverManager.getConnection("jdbc:oracle:thin:@localhost:1521:XE","Student_Performance","Student_Performance");
```
Comment (kabollinger03) [inline src/java/Controller/SendEmail.java:358]: config

File: src/java/Controller/SendEmail.java [hunk 7]
```diff
@@ -182,60 +243,168 @@ else if(count ==1)
...
     	
     	try
-		{
-			Class.forName("oracle.jdbc.driver.OracleDriver"); // Type 4 Driver Pure Java Driver
-			Connection con = DriverManager.getConnection("jdbc:oracle:thin:@localhost:1521:XE","Student_Performance","Student_Performance");
-			Statement st = con.createStatement();
-			ResultSet rs = st.executeQuery("select employee_id, email from employees where class_id = " + "'" + classid + "'");
+	{
+            Class.forName("oracle.jdbc.driver.OracleDriver"); // Type 4 Driver Pure Java Driver
+            Connection con = DriverManager.getConnection("jdbc:oracle:thin:@localhost:1521:XE","Student_Performance","Student_Performance");
```
Comment (kabollinger03) [inline src/java/Controller/SendEmail.java:390]: config

**Review by apj11c — APPROVED** [review]

**Review by apj11c — APPROVED** [review]

**Pull request merged**